### PR TITLE
[chaotic-good] Fix frame fuzzer error.

### PR DESCRIPTION
**Pull request by nanahpang**
The frame fuzzer constructs random bytes and frame header to test serialization and deserialization. There are error cases where the frame header isn't constructed correctly and isn't align with the serialized bytes. Added checks in frame header parser and initialize frame header's flags and header/trailer length in serialization.

### CODE REVIEW

File: src/core/ext/transport/chaotic_good/frame_header.cc [hunk 1]
```diff
@@ -60,9 +60,15 @@ absl::StatusOr<FrameHeader> FrameHeader::Parse(const uint8_t* data) {
   header.flags = BitSet<2>::FromInt(flags);
   header.stream_id = ReadLittleEndianUint32(data + 4);
   header.header_length = ReadLittleEndianUint32(data + 8);
+  if (header.flags.is_set(0) && !(header.header_length > 0)) {
```
Comment (ctiller) [inline src/core/ext/transport/chaotic_good/frame_header.cc:63]: how about:
`if (header.flags.is_set(0) && header.header_length <= 0)`

File: src/core/ext/transport/chaotic_good/frame_header.cc [hunk 2]
```diff
@@ -60,9 +60,15 @@ absl::StatusOr<FrameHeader> FrameHeader::Parse(const uint8_t* data) {
...
+    return absl::InvalidArgumentError("Invalid header length");
+  }
   header.message_length = ReadLittleEndianUint32(data + 12);
   header.message_padding = ReadLittleEndianUint32(data + 16);
   header.trailer_length = ReadLittleEndianUint32(data + 20);
+  if (header.flags.is_set(1) && !(header.trailer_length > 0)) {
```
Comment (ctiller) [inline src/core/ext/transport/chaotic_good/frame_header.cc:69]: here too

File: src/core/ext/transport/chaotic_good/frame_header.cc (hunk 1 above)
Comment (nanahpang) [reply src/core/ext/transport/chaotic_good/frame_header.cc]: Updated, thanks for the suggestion.

File: src/core/ext/transport/chaotic_good/frame_header.cc (hunk 2 above)
Comment (nanahpang) [reply src/core/ext/transport/chaotic_good/frame_header.cc]: Done.

**Review by ctiller — APPROVED** [review]

**Pull request merged**